controllers/productcontroller.go

```
@@ -0,0 +1,74 @@
+	var product entities.Product
+	database.Instance.First(&product, productId)
+	json.NewDecoder(r.Body).Decode(&product)
+	database.Instance.Save(&product)
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(product)
+}
+
+func DeleteProduct(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	productId := mux.Vars(r)["id"]
+	if checkIfProductExists(productId) == false {
```

golangci-lint (1.64.8)

S1002: should omit comparison to bool constant, can be simplified to `!checkIfProductExists(productId)`

(gosimple)